R/zchunk_LA118.hydro.R

```diff
@@ -29,41 +29,177 @@ module_energy_LA118.hydro_DISABLED <- function(command, ...) {
+    # Drop 2020 because it is lower than 2010 in many regions
+    L118.out_EJ_RG3_elec_hydro_fby %>%
+      spread(year, value) %>%
+      left_join_error_no_match(A18.hydro_output, by = "region_GCAM3") %>%
+      gather(year, value, -region_GCAM3) %>%
+      mutate(year = as.numeric(year)) %>%
+      filter(year != 2020) -> # Dropping 2020, as described above
+      L118.out_EJ_RG3_elec_hydro_Y_with_values
+
+    CUSTOM_YEARS <- seq(2010, 2100, 5) # Creating custom year vector to create table in next step
+
+    L118.out_EJ_RG3_elec_hydro_fby %>%
+      select(region_GCAM3) %>%
+      repeat_add_columns(tibble::tibble(year = CUSTOM_YEARS)) %>%
+      left_join(L118.out_EJ_RG3_elec_hydro_Y_with_values, by = c("region_GCAM3", "year")) %>%
+      group_by(region_GCAM3) %>%
+      mutate(value = approx_fun(year, value, rule = 2)) %>% # Interpolation step
+      ungroup() ->
+      L118.out_EJ_RG3_elec_hydro_Y_interp
+
+    # Calculate the growth from the base year in each region_GCAM3. This will be partitioned to countries according to potential
+    L118.out_EJ_RG3_elec_hydro_Y_interp %>%
+      filter( year != 2010) %>% # Deleting 2010 to be used in another column
+      left_join_error_no_match(
+        select(filter(L118.out_EJ_RG3_elec_hydro_Y_interp, year == 2010), -year), # Adding 2010 column so as to add base value to future years
+        by = "region_GCAM3") %>%
+      mutate(value = value.x - value.y) %>%
+      select(-value.x, -value.y) ->
+      L118.growth_EJ_RG3_elec_hydro_Y
+
+    # Calculate the share of the growth potential by each country within its GCAM 3.0 region
+    Hydropower_potential %>%
+      group_by(region_GCAM3) %>%
+      summarise(Growth_potential_EJ_R = sum(Growth_potential_EJ)) %>%
+      filter(!is.na(region_GCAM3)) -> # Getting rid of the NA caused by Kosova not having a RG3 name
+      Hydropower_potential_RG3 # Represents total pie of each country
+
+    Hydropower_potential %>%
+      left_join(Hydropower_potential_RG3, by = "region_GCAM3") %>%
+      mutate(share = Growth_potential_EJ / Growth_potential_EJ_R) %>% # Calculating share
+    # Add future years to table of base-year hydropower generation as base-year output plus region_GCAM3 growth times country-wise share
+      select(iso, share) %>% # Need only share of each iso
+      group_by(iso) %>%
+      summarise(share = sum(share)) -> # Summing up Serbia...was listed twice
+      Hydropower_potential
+
+      # Adding future years
+    L118.out_EJ_ctry_elec_hydro_fby %>%
+      left_join(Hydropower_potential, by = "iso") %>% # Adding share column
+      left_join(L118.growth_EJ_RG3_elec_hydro_Y, by = "region_GCAM3") %>% # Adding RG3 growth
+      mutate(value = value.x + share * value.y) %>% # Base-year output plus RG3 growth times country-wise share
+      select(-share, -value.y) %>%
+      spread(year.x, value.x) %>%
+      spread(year.y, value) %>%
+      gather(year, value, -iso, -region_GCAM3) ->
+      L118.out_EJ_ctry_elec_hydro_Y
```

Comment:
All the `value` variables get really confusing in this section. I'd suggest renaming them _before_ the joins, so that it's clearer what's going on here.